pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -1438,6 +1453,12 @@ var _ = Describe("VirtualMachine Mutator", func() {
 		})
 	})
 
+	It("should default architecture to compiled architecture when not provided", func() {
+		// provide empty string for architecture so that default will apply
+		vmSpec, _ := getVMSpecMetaFromResponse("")
+		Expect(vmSpec.Template.Spec.Architecture).To(Equal(rt.GOARCH))
+	})
+
```

Comment:
Do you have tests for when the defaultArchitecture is set?

Reply:
Let me check on that

Reply:
@rthallisey  Added test for this specifically as well as implemented option #3 discussed above. Are we good to go?